astrobee/config/communications/comms_bridge.config

```diff
@@ -0,0 +1,74 @@
+-- Copyright (c) 2017, United States Government, as represented by the
+-- Administrator of the National Aeronautics and Space Administration.
+--
+-- All rights reserved.
+--
+-- The Astrobee platform is licensed under the Apache License, Version 2.0
+-- (the "License"); you may not use this file except in compliance with the
+-- License. You may obtain a copy of the License at
+--
+--     http://www.apache.org/licenses/LICENSE-2.0
+--
+-- Unless required by applicable law or agreed to in writing, software
+-- distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+-- WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+-- License for the specific language governing permissions and limitations
+-- under the License.
+
+-- param for each bridge component use the following params --
+--  ABREV = First Letters of Component Ex: ros_string_rapid_text_msg = RSRTM --
+--   use_ABREV        bool:   turn this component on? --
+--   sub_topic_ABREV  string: [ROS: subscribe topic] or [RAPID: subscribe topic suffix] --
+--   pub_topic_ABREV  string: [ROS: publish topic] or [RAPID: publish topic suffix] --
+--   queue_size_ABREV int:    queue size --
+
+require "context"
+
+-- Enable DDS communications on start --
+-- If false, a trigger call will be needed to start communications --
+started = false;
+
+-- Rapid subscribers --
+rapid_sub_names = {"Bumble", "Honey", "Queen", "Bsharp", "Wannabee"}
```

Comment:
I'm not entirely sure what you are asking but thank you for asking it as just realize I have a bit of undesired behavior that I need to fix. As you know, the relay code converts a ros message into an advertisement info message (only the first time it sees the message type) and the content message. I was just publishing the messages on dds topics "generic_comms_advertisement_info" and "generic_comms_content". However this means all connected robots will receive all topics so I will change this to append the receive robot's name to it.

Reply:
Ha, glad you caught that.

But I was talking about a different issue: when a ROS message is relayed from robot A to robot B, how do we specify the relationship between the topic used locally on A and the topic of the relayed message that the bridge publishes on B?

In my draft spec for the config file, the idea was that you could either manually specify both the "from" and "to" topics in the relay entry, or if you prefer, you could specify only the "from" topic and it would auto-generate the "to" topic using a default template string.

I think I understand now that your concept is to hard-code the relationship between "from" and "to" topics because we need it to work a certain way for compatibility with the topic naming convention for multi-robot sims. That's probably ok, but I think it should be documented here, something like:

"The topic `$topic` you specify in this config is a suffix of the full topic name and must not start with `/`. It will be combined with different prefixes based on the context. When relaying from `$from_robot` to `$to_robot`, the `from` side of the bridge subscribes to the local topic `/$topic` and the `to` side of the bridge publishes the namespaced global topic `/$from_robot/$topic`. This convention is designed for compatibility with how we run multi-robot sims. It allows us to specify that a ROS node running on `$to_robot` should communicate with ROS nodes on `$from_robot` (instead of local ROS nodes) by launching it with the command-line argument `-ns $from_robot`."

I also have a trivial suggestion that in each of these relay entries, we should call the main field `topic` instead of `name`, just to be self-documenting.

Reply:
I apologize, I opened your draft spec for the config file in a tab and completely forgot to read it before working on the bridge. I love your idea of having an optional "to" topic and generating the "to" topic by appending the robot name to the beginning of the topic if the "to" topic isn't specified. I plan to add that this week and I also plan to thoroughly document this before putting the pr in official review. Further, I will change name to topic. :)